### DISCUSSION TITLE
Copilot workflow editor using SK planner

### DISCUSSION THREAD

**Discussion by Woland2k** [microsoft/semantic-kernel#5715] [Q&A] [answered]
Not a feature request but rather a question.

Was Copilot Studio "edit with copilot" functionality for editing Topics developed using SK? If not, would it be a right approach to use planner to create something similar (basically I have preexisting nodes that I'd like engine to use to construct a logical tree).

![image](https://github.com/microsoft/semantic-kernel/assets/1566470/af1b2278-ab4e-47d7-986d-739223ab119d)

**Comment by madsbolaris** [comment, marked as answer]
If you want to recreate similar logic within Semantic Kernel, the first thing you should do, is model each one of the nodes as a `KernelFuncion`. From there, you can use standard C# (or Python/Java) code to sequentially call each of your functions and passing the input/output from one function to the other.

Once you want the AI to _automatically_ call your functions, you can use automatic function calling within Semantic Kernel so your LLM can automatically invoke any of your plugins.

**Reply by alexandreakdeniz-soc** [reply]
Hello,

Your answer is very insteresting, how would you manage branches ?
If there is two possibles nodes you could then go on, how would you know on which one to go ?

**Reply by stephane-liwerant** [reply]
Hello, 
Do you mean that Semantic Kernel will not help with managing specific predefined plan? 
If yes, how do you manage the new chat messages you may want to generate along the plan, for additional info for example?

**Comment by pchao04** [comment]
As far as I learn, Copilor can intergrate with Power Automate, then Power Automate can integrate with Sematic Kernel ?
